lib/widgets/emoji_reaction.dart

```diff
@@ -614,3 +627,411 @@ class EmojiPickerListEntry extends StatelessWidget {
+                  count.toString()), // TODO(i18n) number formatting?
+              ])))));
+
+    return Semantics(
+      role: SemanticsRole.tab,
+      onDidGainAccessibilityFocus: () => _scrollIntoView(context),
+
+      // I *think* we're following the doc with this but it's hard to tell;
+      // I've only tested on iOS and I didn't notice a behavior change.
+      controlsNodes: {ViewReactionsUserList.semanticsIdentifier},
+
+      selected: selected,
+      label: zulipLocalizations.seeWhoReactedSheetEmojiNameWithVoteCount(emojiName, count),
+      onTap: onRequestSelect,
+      child: ExcludeSemantics(
+        child: result));
+  }
+}
+
+
+@visibleForTesting
+class ViewReactionsUserList extends StatelessWidget {
+  const ViewReactionsUserList(this.pageContext, {
+    super.key,
+    required this.messageId,
+    required this.reactionType,
+    required this.emojiCode,
+    required this.emojiName,
+  });
+
+  final BuildContext pageContext;
+  final int messageId;
+  final ReactionType reactionType;
+  final String emojiCode;
+  final String emojiName;
+
+  static const semanticsIdentifier = 'view-reactions-user-list';
+
+  @override
+  Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final designVariables = DesignVariables.of(context);
+
+    final message = store.messages[messageId];
+
+    final userIds = message?.reactions?.aggregated.firstWhereOrNull(
+      (x) => x.reactionType == reactionType && x.emojiCode == emojiCode
+    )?.userIds.toList();
+
+    // (No filtering of muted or deactivated users.)
+
+    if (userIds == null) {
+      // This reaction lost all its votes, or the message was deleted.
+      return SizedBox.shrink();
+    }
+
+    // TODO sort userIds?
+
+    Widget result = InsetShadowBox(
+      top: 8, bottom: 8,
+      color: designVariables.bgContextMenu,
+      child: SizedBox(
+        height: 400, // TODO(design) tune
+        child: ListView.builder(
+          padding: EdgeInsets.symmetric(vertical: 8),
+          itemCount: userIds.length,
+          itemBuilder: (context, index) =>
+            ViewReactionsUserItem(context, userId: userIds[index]))));
+
+    return Semantics(
+      identifier: semanticsIdentifier, // See note on `controlsNodes` on the tab.
+      label: zulipLocalizations.seeWhoReactedSheetUserListLabel(emojiName, userIds.length),
+      role: SemanticsRole.tabPanel,
+      container: true,
+      child: result);
+  }
+}
+
+@visibleForTesting
+class ViewReactionsUserItem extends StatelessWidget {
+  const ViewReactionsUserItem(this.pageContext, {
+    super.key,
+    required this.userId,
+  });
+
+  final BuildContext pageContext;
+  final int userId;
+
+  void _onPressed() {
+    // Dismiss the action sheet.
+    Navigator.pop(pageContext);
+
+    Navigator.push(pageContext,
+      ProfilePage.buildRoute(context: pageContext, userId: userId));
```

Comment:
I think the reason this works even though this `pageContext` isn't really a page context is that there's no `await` here.

The reason our action-sheet buttons often need a real page context is that they'll dismiss the action sheet, then do something that takes time like a network request, and then after that completes they need to use a context to do something with the result, or to show an error. So if the request takes longer than the few hundred ms of the dismiss animation, the action sheet's own context may be unmounted by that point.

For just synchronously acting on the navigation like this, the widget's own context is fine.